src/box/box.cc

```diff
@@ -1375,10 +1375,22 @@ box_process1(struct request *request, box_tuple_t **result)
 	struct space *space = space_cache_find(request->space_id);
 	if (space == NULL)
 		return -1;
+	struct engine *engine = (struct engine *)space->engine;
 	if (!space_is_temporary(space) &&
 	    space_group_id(space) != GROUP_LOCAL &&
 	    box_check_writable() != 0)
 		return -1;
+
+	if(engine && engine->id == 0) {
```

Comment:
engine really can be null? Removed this check and box/engine suits seem to pass.

Reply:
You are are right. Space->engine sets when space created